packages/cardano-graphql-services/src/DataProjection/blockHandlers/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import { Asset, BigIntMath } from '@cardano-sdk/core';
+import { AssetMetadata } from '../../MetadataClient/types';
+import { Schema, isAlonzoBlock, isMaryBlock } from '@cardano-ogmios/client';
+
+export const getBlockType = (block: Schema.Block) => {
+  let b: Schema.BlockMary | Schema.BlockAlonzo | undefined;
+  if (isAlonzoBlock(block)) {
+    b = block.alonzo as Schema.BlockAlonzo;
+  } else if (isMaryBlock(block)) {
+    b = block.mary as Schema.BlockMary;
+  }
+  return b;
+};
+
+export const mapMetadata = (serverAssetMetadata: AssetMetadata): Asset.TokenMetadata => ({
```

Comment:
Since `mapBlock` is used for mutations, we should be mapping to our [graphql schema types](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/cardano-graphql-services/src/Schema/types/Asset/Asset.ts) instead of core types